### PR TITLE
Some missing specs and minor cleanup

### PR DESCRIPTION
**Pull request by chetanism**
o Added few missing specs
o Export adapters from index
o Moved indexSpec to integration directory
o Added simple indexSpec that checks exports

### CODE REVIEW

**Comment by chetanism** [comment]
@aexmachina, I see you have added `BookshelfAdapter.init` and `MongooseAdapter.init` methods. IMHO, I don't like them much. I suppose they exist in support of previous version where you could just require an adapter and invoke it to set it as default. However, I have a couple of points against them:

1] They make adapter depend on `factory` object.
2] They just affect the default `factory` object exposed by the library.
3] It's hard to unit test these `init` methods.

When users upgrade to v4, they will need to make a few changes. I suppose setting the default adapter could be one of them. What do you think?

**Comment by aexmachina** [comment]
Okay fair enough. I was just trying to make using an adapter a one-liner in the test setup. But I agree that it's better to be explicit. Happy to remove these.

**Comment by chetanism** [comment]
ok @aexmachina, I'll just push another commit to remove the init methods. Rest, if it looks good you can then merge the PR!

**Comment by chetanism** [comment]
@aexmachina, I have pushed changes to remove the `init` methods.

File: src/adapters/BookshelfAdapter.js [hunk 1]
```diff
@@ -8,6 +7,4 @@ export default class BookshelfAdapter extends DefaultAdapter {
   }
 }
 
-BookshelfAdapter.init = function init(factoryNames) {
-  return factory.setAdapter(new BookshelfAdapter(), factoryNames);
-};
+export { BookshelfAdapter };
```
Comment (aexmachina) [inline src/adapters/BookshelfAdapter.js]: It's the default export, no need to do a named export.

**Comment by aexmachina** [comment]
Apart from that, this is ready to merge. Feel free to merge when that's fixed.

**Comment by chetanism** [comment]
Done!

**Pull request merged**